packages/peregrine/lib/mixins/MyAccount/useMyAccount.js

```diff
@@ -0,0 +1,27 @@
+import { useCallback } from 'react';
+import { useUserContext } from '../../context/user';
+
+const DEFAULT_TITLE = 'My Account';
+const UNAUTHED_TITLE = 'Signing Out';
+const UNAUTHED_SUBTITLE = 'Please wait...';
+
+const useMyAccount = props => {
+    const { signOut } = props;
+    const [{ currentUser }] = useUserContext();
+    const { email, firstname, lastname } = currentUser;
+    const name = `${firstname} ${lastname}`.trim() || DEFAULT_TITLE;
+    const title = email ? name : UNAUTHED_TITLE;
+    const subtitle = email ? email : UNAUTHED_SUBTITLE;
+
+    const handleSignOut = useCallback(() => {
+        signOut({ history: window.history });
```

Comment:
We'll want to use the router history, which we can hopefully get from context. There's not much point in passing around window properties anyway.

Reply:
Sure -- I just didn't change the api for it. This is a zero-functional change.